src/community/web-resources/src/main/java/org/geoserver/web/resources/PageResourceBrowser.java

```diff
@@ -0,0 +1,516 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.web.resources;
+
+import org.apache.wicket.Component;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.markup.html.AjaxLink;
+import org.apache.wicket.behavior.AttributeAppender;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.link.AbstractLink;
+import org.apache.wicket.markup.html.link.Link;
+import org.apache.wicket.model.Model;
+import org.apache.wicket.request.handler.resource.ResourceStreamRequestHandler;
+import org.apache.wicket.request.resource.ContentDisposition;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.web.GeoServerSecuredPage;
+import org.geoserver.web.treeview.TreeNode;
+import org.geoserver.web.treeview.TreeView;
+import org.geoserver.web.wicket.GeoServerDialog;
+import org.geoserver.web.wicket.ParamResourceModel;
+import org.geoserver.web.wicket.GeoServerDialog.DialogDelegate;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * The ResourceBrowser page.
+ * 
+ * @author Niels Charlier
+ *
+ */
+public class PageResourceBrowser extends GeoServerSecuredPage {
+    
+    private static final long serialVersionUID = 3979040405548783679L;
+        
+    /**
+     * Behaviour for disabled button
+     */
+    private static final AttributeAppender DISABLED_BEHAVIOR = new AttributeAppender("class", new Model<String>("disabled"), " ");
+
+    /**
+     * The extension that are recognised as simple text resources (and can be edited with simple text editor).
+     */
+    private static final String[] TEXTUAL_EXTENSIONS = new String[] {"txt", "properties", "info", "xml", "sld", "rst", "log"};
```

Comment:
There are many more text extensions that might benefit, including palettes, freemarker templates, HTML, and JavaScript. Yes, you have turned GeoServer into a web CMS.  :-)

For example, in the GeoServer release data directory, a crude attempt to extract unique suffices:
```
$ find . -exec file {} \; | grep text | perl -p -e 's/:.*//' | perl -p -e 's/.*\.//' | sort | uniq
/security/filter/contextAsc
/security/filter/contextNoAsc
/security/masterpw/default/passwd
asc
cfg
css
ftl
html
js
meta
pal
pgw
prj
properties
rst
sld
tfw
txt
url
xml
xsd
```